### PR TITLE
Gcc 9 for 0.10 branch

### PR DESCRIPTION
**Pull request by yoe**
This gets all the changes that are also on #1578, but on top of the 0.10 branch.

It should actually work, as opposed to #1568...

### CODE REVIEW

File: plugins/dummy/DummyPortTest.cpp [hunk 1]
```diff
@@ -662,7 +662,22 @@ void DummyPortTest::testParamDescription() {
   const string description("Code Version");
   size_t str_len = std::min(sizeof(param_description.description),
                             description.size());
+/* Some versions of GCC 9 claim that the strncpy overflows the target
```
Comment (peternewman) [inline plugins/dummy/DummyPortTest.cpp]: Nit, we generally have the /* clear at the start of a block comment, with the comment starting on the next line.

File: plugins/usbpro/RobeWidgetDetector.h [hunk 2]
```diff
@@ -56,6 +56,12 @@ class RobeWidgetInformation {
   uint8_t hardware_version;
   uint8_t software_version;
   uint8_t eeprom_version;
+  void operator=(const RobeWidgetInformation& other) {
+      hardware_version = other.hardware_version;
+      software_version = other.software_version;
+      eeprom_version = other.eeprom_version;
+      uid = other.uid;
```
Comment (peternewman) [inline plugins/usbpro/RobeWidgetDetector.h]: Nit, would it make sense for the order of these to match the constructor that passes in other? I'm not quite sure what your order is, apart from basic types first, but not even alphabetical.

File: plugins/dummy/DummyPortTest.cpp [hunk 3]
```diff
@@ -662,7 +662,22 @@ void DummyPortTest::testParamDescription() {
...
+ * string. While that's (kindof) true, all that happens is that we drop
+ * the NUL byte, but that's on purpose.
+
+ * Disable the warning to avoid it being an issue, but only for this
+ * line.
+ *
+ * Also, work around clang producing an erorr on the
```
Comment (peternewman) [inline plugins/dummy/DummyPortTest.cpp]: Nit, typo error.

**Comment by peternewman** [comment]
Did you get a chance to look at these comments @yoe?

File: plugins/dummy/DummyPortTest.cpp (hunk 3 above)
Comment (peternewman) [reply plugins/dummy/DummyPortTest.cpp]: Done.

File: plugins/dummy/DummyPortTest.cpp (hunk 1 above)
Comment (peternewman) [reply plugins/dummy/DummyPortTest.cpp]: Done.

**Comment by yoe** [comment]
I'm afraid I'm going to have to punt on this.

I was hoping to find the time to finish this at some point, and actually did try to get anywhere a while back, but I keep running into "ola expects older versions of software" type of issues. Meanwhile, GCC10 is about to release, and we still don't compile with GCC9.

I might look at fixing this for the master branch again, but 0.10 is just too old.

If someone else want to take this out of my hands, I'd be happy to hand it to them, otherwise I'll close this in a few days.

**Comment by peternewman** [comment]
> I'm afraid I'm going to have to punt on this.

Thanks for your efforts @yoe , I completely understand.

> I was hoping to find the time to finish this at some point, and actually did try to get anywhere a while back, but I keep running into "ola expects older versions of software" type of issues. Meanwhile, GCC10 is about to release, and we still don't compile with GCC9.

:cry: 

> I might look at fixing this for the master branch again, but 0.10 is just too old.

I think most of the protobuf etc fixes have now been pulled back into 0.10 too, so it should be unencumbered by this.

> If someone else want to take this out of my hands, I'd be happy to hand it to them, otherwise I'll close this in a few days.

Would you mind leaving it open? Well I guess you already have. I might try a bit of hacking around at it, or someone else could pull your changes in to their own branch.

File: plugins/usbpro/RobeWidgetDetector.h (hunk 2 above)
Comment (peternewman) [reply plugins/usbpro/RobeWidgetDetector.h]: Done.

**Comment by peternewman** [comment]
> Thanks for this. A few comments, mostly minor nits.
> 
> Also this still doesn't seem to work on Clang, the log has:
> 
> ```
> ../../common/testing/TestUtils.cpp:68: 20: 0xfd != 0x43 (  != C)  ## MISMATCH
> ../../common/testing/TestUtils.cpp:68: 21: 0x7f != 0x6f (  != o)  ## MISMATCH
> ../../common/testing/TestUtils.cpp:68: 22: 0x0 != 0x64 (  != d)  ## MISMATCH
> ../../common/testing/TestUtils.cpp:68: 23: 0x0 != 0x65 (  != e)  ## MISMATCH
> ../../common/testing/TestUtils.cpp:68: 24: 0x62 != 0x20 (b !=  )  ## MISMATCH
> ../../common/testing/TestUtils.cpp:68: 25: 0x66 != 0x56 (f != V)  ## MISMATCH
> ../../common/testing/TestUtils.cpp:68: 26: 0x40 != 0x65 (@ != e)  ## MISMATCH
> ../../common/testing/TestUtils.cpp:68: 27: 0x0 != 0x72 (  != r)  ## MISMATCH
> ../../common/testing/TestUtils.cpp:68: 28: 0x0 != 0x73 (  != s)  ## MISMATCH
> ../../common/testing/TestUtils.cpp:68: 29: 0x0 != 0x69 (  != i)  ## MISMATCH
> ../../common/testing/TestUtils.cpp:68: 30: 0x0 != 0x6f (  != o)  ## MISMATCH
> ../../common/testing/TestUtils.cpp:68: 31: 0x0 != 0x6e (  != n)  ## MISMATCH
> ```

Hopefully fixed. :crossed_fingers: 

> LGTM's static analysis also suggests the same missing constructor/assignment pair for OSCTarget and GenericSocketAddress. I'm not sure why they're not flagged by this but do you fancy adding them too:
> https://lgtm.com/projects/g/OpenLightingProject/ola/?mode=list&lang=cpp

Done.

**Comment by peternewman** [comment]
@yoe this now seems to pass all the Travis tests after a very minor tweak to your change. I'm tempted to just merge it as is unless you've still got access to your GCC-9 environment and fancy giving it a little bit more of a shakedown?

**Comment by yoe** [comment]
I don't think I still have that access. Also, if I did, it would be GCC10 rather than GCC9 :-)

Please merge, we can move on from there.

Thanks,

**Review by peternewman — APPROVED** [review]
LGTM as we've mutually fixed all the outstanding issues.

**Pull request merged**